### DISCUSSION TITLE
Can we create a light theme?

### DISCUSSION THREAD

**Discussion by thomasnield** [3b1b/manim#1738] [Q&A] [answered]
Hey all, 

I've been exploring Manim off and on for awhile and finally hunkered down today. I think this is a great visualization library and I really want to use it for my educational tools. 

Is it possible we can create a light theme though? I write books and the only thing keeping me from using this over Desmos, matplotlib, and Plotly for illustrations is it's dark theming. While the dark theming is cool for videos, it is not paper-friendly if I want to generate static images. While the colors can be overridden, it would be nice to standardize a light theme. 

Just a thought and thank you for all this fantastic work.

**Comment by TonyCrane** [comment, marked as answer]
ManimGL can make light color themes videos/pictures, because all colors can also be changed.
If you want to change the background color, you can change the `default_config.yml` file in the `manimlib/` folder or the `custom_config.yml` file in the current directory. Or just run `manimgl --config` and it will guide you through these configurations.
To change the color of the `Mobject`, you can directly pass in the parameters such as `color`, `stroke_color`, `fill_color`, `stroke_opacity`, `fill_opacity`, `background_stroke_color`, `background_stroke_opacity`, etc.
Although the default color does not provide an option for a light theme, which will cause each `Mobject` to need to modify the color, but if you feel repetitive, you can write the style into a dictionary, or inherit and override the class, like:
```python
style = {"fill_color": BLACK, "background_stroke_color": WHITE}
tex = Tex("a^2+b^2=c^2", **style)
# or 
class Tex_(Tex):
    CONFIG = {
        "fill_color": BLACK,
        "background_stroke_color": WHITE,
    }
```

**Reply by thomasnield** [reply]
Okay thank you, I will play with this.

**Comment by Alexander-Nasuta** [comment]
Hi everyone,

Theming is really important to me, so I recently wrote a Manim plugin that lets you override the default colors using terminal color palettes, specifically from the iTerm2 collection.

Here’s how it works:
You call an apply_theme function and pass in the name of an iTerm2 color palette. If the palette isn’t already in your media folder, the plugin will automatically download it from GitHub and update Manim's default colors accordingly.

One light theme I really like is Monokai Pro Light—here’s an example of how it looks:
<img width="1920" height="1080" alt="ThemeGif_ManimCE_v0 19 0 monokai" src="https://github.com/user-attachments/assets/c5500b84-62ab-4c51-b264-5b42418d2999" />

<img width="1920" height="1080" alt="ThemeGif_ManimCE_v0 19 0 monkai" src="https://github.com/user-attachments/assets/ca3936b7-30bd-4b7d-851e-18e8c1c02060" />


Here is a minimal example:

install the plugin via pip
```
   pip install manim-themes
```

And here is a minimal Scene:
```
import manim as m

# import the apply_theme function from the manim_themes package
from manim_themes.manim_theme import apply_theme


class MinimalThemeExample(m.Scene):

    def setup(self):
        theme = "Monokai Pro Light" # select a theme from https://iterm2colorschemes.com
        apply_theme(manim_scene=self, theme_name=theme, light_theme=True) # use the theme

    def construct(self):
        my_text = m.Text("Hello World")
        maroon_text = m.Text("I use Manim BTW", color=m.MAROON)
        maroon_text.next_to(my_text, m.DOWN)

        text_group = m.VGroup(my_text, maroon_text).move_to(m.ORIGIN)

        self.play(m.FadeIn(text_group))


if __name__ == '__main__':
    import os
    from pathlib import Path

    FLAGS = "-pqm -s"
    SCENE = "MinimalThemeExample"

    file_path = Path(__file__).resolve()
    os.system(f"manim {Path(__file__).resolve()} {SCENE} {FLAGS}")
```

This will result in this Scene:

<img width="1280" height="720" alt="MinimalThemeExample_ManimCE_v0 19 0 MonokaiExample" src="https://github.com/user-attachments/assets/ad12dec6-4392-4460-8b1a-ce6f863234bd" />


The plugin is documented here:
https://manim-themes.readthedocs.io/en/latest/index.html

Repo: https://github.com/Alexander-Nasuta/manim-themes

Note: This plugin is for **Manim Community Edition** (ManimCE), not ManimGL.

I’m sharing this here because it might be useful for others who are looking for a standardized light theme, even though this isn't the ManimCE repo.

Hope it helps!
